### PR TITLE
Refactor userـldap app commands

### PR DESCRIPTION
**Pull request by fsamapoor**
## Summary
I have made some adjustments to the `apps/user_ldap/lib/Command` classes to improve the code readability.

The improvements in this PR include but are not limited to:

- Using PHP8's constructor property promotion
- Adding return types
- Replacing return code integers with more readable strings.
- Using early returns

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits

### CODE REVIEW

File: apps/user_ldap/lib/Command/CreateEmptyConfig.php [hunk 1]
```diff
@@ -67,6 +62,6 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 			$prose = 'Created new configuration with configID ';
 		}
 		$output->writeln($prose . "{$configPrefix}");
-		return 0;
+		return self::SUCCESS;
```
Comment (Fenn-CS) [inline apps/user_ldap/lib/Command/CreateEmptyConfig.php:65]: Why is a method annotated with void returning something?

File: apps/user_ldap/lib/Command/ResetUser.php [hunk 2]
```diff
@@ -36,25 +36,15 @@
...
+		private UserPluginManager $pluginManager,
 	) {
-		$this->dui = $dui;
-		$this->userManager = $userManager;
-		$this->pluginManager = $pluginManager;
 		parent::__construct();
 	}
 
-	protected function configure() {
+	protected function configure(): void {
```
Comment (Fenn-CS) [inline apps/user_ldap/lib/Command/ResetUser.php:47]: Same comment as above, this method returns values. Can't be void.

**Review by nfebe — COMMENTED** [review]
Looks like some void-annotated methods do actually return values?

**Comment by fsamapoor** [comment]
> Looks like some void-annotated methods do actually return values?

Hello @Fenn-CS, Thank you for reviewing the PR.

No method with the return type of `void` returns any values. The git diff seems like the return belongs to the method with a `void` return type, but it actually does not.

These two lines do not belong to the same method:

image

Here is the expanded diff:

image

File: apps/user_ldap/lib/Command/CreateEmptyConfig.php (hunk 1 above)
Comment (fsamapoor) [reply apps/user_ldap/lib/Command/CreateEmptyConfig.php:65]: The git diff seems confusing. No value is returned form a method with the `void` return type. I explained it in a separate comment.

File: apps/user_ldap/lib/Command/ResetUser.php (hunk 2 above)
Comment (fsamapoor) [reply apps/user_ldap/lib/Command/ResetUser.php:47]: The git diff seems confusing. No value is returned from a method with the `void` return type. I explained it in a separate comment.

**Review by nfebe — APPROVED** [review]
Thanks Faraz!

**Review by blizzz — APPROVED** [review]

File: apps/user_ldap/lib/Command/Search.php [hunk 3]
```diff
@@ -87,22 +81,20 @@ protected function configure() {
...
+		$message = match (true) {
+			($limit < 0) => 'limit must be 0 or greater',
+			($offset < 0) => 'offset must be 0 or greater',
+			($limit === 0 && $offset !== 0) => 'offset must be 0 if limit is also set to 0',
+			($offset > 0 && ($offset % $limit !== 0)) => 'offset must be a multiple of limit',
+			default => '',
+		};
+
+		if (!empty($message)) {
+			throw new \InvalidArgumentException($message);
```
Comment (come-nc) [inline apps/user_ldap/lib/Command/Search.php:97]: Could you revert this part?
I really see no point apart from using the shiny new match keyword. It’s harder to read.

File: apps/user_ldap/lib/Command/UpdateUUID.php [hunk 4]
```diff
@@ -49,45 +49,29 @@ class UuidUpdateReport {
...
-		$this->dn = $dn;
-		$this->isUser = $isUser;
-		$this->state = $state;
-		$this->oldUuid = $oldUuid;
-		$this->newUuid = $newUuid;
+	public function __construct(
+		public string $id = '',
+		public string $dn = '',
+		public bool $isUser = true,
+		public int $state = self::UNCHANGED,
```
Comment (come-nc) [inline apps/user_ldap/lib/Command/UpdateUUID.php:56]:
```suggestion
		public string $id,
		public string $dn,
		public bool $isUser,
		public int $state,
```
These should stay mandatory parameters

**Review by come-nc — COMMENTED** [review]
I’m not sure why there is so much hate towards `else` statements, I find things clearer with them that without. Especially the `continue` in the loop instead of a simple `else` looks convoluted.
I like the strong typing, constructor promotion and constant usage in returns though.

Please revert at least the `match (true)`.

**Comment by fsamapoor** [comment]
> I’m not sure why there is so much hate towards `else` statements, I find things clearer with them that without. Especially the `continue` in the loop instead of a simple `else` looks convoluted. I like the strong typing, constructor promotion and constant usage in returns though.
> 
> Please revert at least the `match (true)`.

Thank you for taking the time to review the changes Come. This was one of my older PRs and I appreciate your suggestions.

File: apps/user_ldap/lib/Command/Search.php [hunk 5]
```diff
@@ -87,11 +81,10 @@ protected function configure() {
 
 	/**
 	 * Tests whether the offset and limit options are valid
-	 * @param int $offset
-	 * @param int $limit
+	 *
 	 * @throws \InvalidArgumentException
 	 */
-	protected function validateOffsetAndLimit($offset, $limit) {
+	protected function validateOffsetAndLimit($offset, $limit): void {
```
Comment (come-nc) [inline apps/user_ldap/lib/Command/Search.php:87]:
```suggestion
	protected function validateOffsetAndLimit(int $offset, int $limit): void {
```

**Review by come-nc — APPROVED** [review]

**Pull request merged**